python/generateTemplate.py

```diff
@@ -567,87 +560,13 @@ def load_bandpass(filter_name='g', wavelength_step=None, use_mirror=True, use_le
-
-            def calc_eff_wavelen(self, wavelength_min=None, wavelength_max=None):
-                """Calculate effective wavelengths for filters.
-
-                Parameters
-                ----------
-                wavelength_min : float, optional
-                    Starting wavelength, in nm
-                wavelength_max : float, optional
-                    End wavelength, in nm
-
-                Returns
-                -------
-                Returns the weighted average wavelength within the range given, taken over the bandpass.
-                """
-                if self.phi is None:
-                    self.sbTophi()
-                if wavelength_min is None:
-                    wavelength_min = np.min(self.wavelen)
-                if wavelength_max is None:
-                    wavelength_max = np.max(self.wavelen)
-                w_inds = (self.wavelen >= wavelength_min) & (self.wavelen <= wavelength_max)
-                effwavelenphi = (self.wavelen[w_inds]*self.phi[w_inds]).sum()/self.phi[w_inds].sum()
-                return effwavelenphi
-
-            def calc_bandwidth(self):
-                f0 = constants.speed_of_light/(self.wavelen_min*1.0e-9)
-                f1 = constants.speed_of_light/(self.wavelen_max*1.0e-9)
-                f_cen = constants.speed_of_light/(self.calc_eff_wavelen()*1.0e-9)
-                return(f_cen*2.0*(f0 - f1)/(f0 + f1))
-
+        Returns a BandpassHelper object that has an interface similar to lsst.sims.photUtils.Bandpass.
         """
-        Define the wavelength range and resolution for a given ugrizy band.
-        These are defined in case the LSST filter throughputs are not used.
-        """
-        band_dict = {'u': (324.0, 395.0), 'g': (405.0, 552.0), 'r': (552.0, 691.0),
-                     'i': (818.0, 921.0), 'z': (922.0, 997.0), 'y': (975.0, 1075.0)}
-        band_range = band_dict[filter_name]
-        bandpass = BandpassMod(wavelen_min=band_range[0], wavelen_max=band_range[1],
-                               wavelen_step=wavelength_step)
-        throughput_dir = getPackageDir('throughputs')
-        lens_list = ['baseline/lens1.dat', 'baseline/lens2.dat', 'baseline/lens3.dat']
-        mirror_list = ['baseline/m1.dat', 'baseline/m2.dat', 'baseline/m3.dat']
-        atmos_list = ['atmos/atmos_11.dat']
-        detector_list = ['baseline/detector.dat']
-        filter_list = ['baseline/filter_' + filter_name + '.dat']
-        component_list = []
-        if use_mirror:
-            component_list += mirror_list
-        if use_lens:
-            component_list += lens_list
-        if use_atmos:
-            component_list += atmos_list
-        if use_detector:
-            component_list += detector_list
-        if use_filter:
-            component_list += filter_list
-        bandpass.readThroughputList(rootDir=throughput_dir, componentList=component_list)
-        # Calculate bandpass phi value if required.
-        if bandpass.phi is None:
```

Comment:
I take it this large amount of code was moved to BandPassHelper?

Reply:
Correct.